src/arch-clap.ts

```diff
@@ -0,0 +1,518 @@
+/* eslint-disable no-console, global-require, no-magic-numbers, max-statements */
+import * as Path from "path";
+import * as xsh from "xsh";
+import * as Fs from "fs";
+import * as filterScanDir from "filter-scan-dir";
+import * as xclap from "xclap";
+import * as _ from "lodash";
+
+const typeScriptDevDeps = {
+  // code coverage
+  "@istanbuljs/nyc-config-typescript": "^1.0.1",
+  "source-map-support": "^0.5.16",
+  // types for node.js
+  "@types/node": "^13.7.6",
+  // compilers
+  "ts-node": "^8.6.2",
+  typescript: "^3.8.3"
+};
+
+const eslintDevDeps = {
+  "babel-eslint": "^10.1.0",
+  eslint: "^6.8.0",
+  "eslint-config-walmart": "^2.2.1",
+  "eslint-plugin-filenames": "^1.1.0",
+  "eslint-plugin-jsdoc": "^21.0.0"
+};
+
+const eslintTSDevDeps = {
+  // eslint typescript deps
+  "@typescript-eslint/eslint-plugin": "^2.21.0",
+  "@typescript-eslint/parser": "^2.21.0"
+};
+
+/**
+ * User configurable options for @xarc/module-dev tasks
+ */
+export type XarcModuleDevOptions = {
+  /** force terminal colors in output - *default* `true` */
+  forceColor?: boolean;
+  /** Specify a XClap instance to use - *default* `require("xclap")` */
+  xclap?: object;
+  /** turn off/on linting tasks (using eslint) - *default* `true` */
+  enableLinting?: boolean;
+  /** Specify typescript config to override the default one */
+  tsConfig?: object;
+};
+
+/**
+ * setup PATH
+ */
+function setupPath(): void {
+  xsh.envPath.addToFront(Path.resolve("node_modules/.bin"));
+  xsh.envPath.addToFront(Path.join(__dirname, "../node_modules/.bin"));
+}
+
+/**
+ * Sort property keys of an object
+ *
+ * @param obj object
+ * @returns obj
+ */
+function sortObjKeys(obj: any): any {
```

Comment:
Never use `any` unless you absolutely have to. In this case, it's not needed.

Since the object being passed in is the same type as the object being returned, we can use a generic. We also want to limit this function to take objects (not primitives), so we'll want to make sure the input extends `{}`. With these changes, we get:

```ts

function sortObjKeys<T extends {}>(obj: T): T {
  return (Object.keys(obj) as (keyof T)[])  // The built in type of `Object.keys` returns `string[]` instead of the exact string values of each key, so we need to cast
    .sort()
    .reduce((newObj, key) => {
      newObj[key] = obj[key];
      return newObj;
    }, {} as T); // A cast here is used to tell TS that this object will eventually be T (when we return)
}
```

BTW, just like `any`, casts (`as`) should be used sparingly. Here they were necessary, but that's not often.

Reply:
tsc doesn't complain `sortObjKeys(false)` with type from `{}`.  Is there a strict mode that needs to be enabled?